pkg/kubectl/cmd/rollout/rollout_history.go

```diff
@@ -98,27 +98,17 @@ func RunHistory(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, args []st
 		if err != nil {
 			return err
 		}
-		historyInfo, err := historyViewer.History(info.Namespace, info.Name)
+		historyInfo, err := historyViewer.ViewHistory(info.Namespace, info.Name, revision)
 		if err != nil {
 			return err
 		}
 
-		if revisionDetail > 0 {
-			// Print details of a specific revision
-			template, ok := historyInfo.RevisionToTemplate[revisionDetail]
-			if !ok {
-				return fmt.Errorf("unable to find revision %d of %s %q", revisionDetail, mapping.Resource, info.Name)
-			}
-			fmt.Fprintf(out, "%s %q revision %d\n", mapping.Resource, info.Name, revisionDetail)
-			kubectl.DescribePodTemplate(template, out)
-		} else {
-			// Print all revisions
-			formattedOutput, printErr := kubectl.PrintRolloutHistory(historyInfo, mapping.Resource, info.Name)
-			if printErr != nil {
-				return printErr
-			}
-			fmt.Fprintf(out, "%s\n", formattedOutput)
+		header := fmt.Sprintf("%s %q history viewed", mapping.Resource, info.Name)
+		if revision > 0 {
+			header = fmt.Sprintf("%s (revision: %d)", header, revision)
```

Comment:
In the case when `--revision` is specified, the header `deployments "xxx" history viewed (revision: xx)` looks a bit weird to me. This message is useful when we create or update something without printing it (e.g. `kubectl create`), but isn't appropriate as a header in this case IMO. 

How about just `deployments "xxx" revision xx` or something similar? The user would just need to know which deployment and which revision.